command/agent/auth/azure/azure.go

```diff
@@ -63,11 +65,29 @@ func NewAzureAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 		return nil, errors.New("could not convert 'resource' config value to string")
 	}
 
+	objectIDRaw, ok := conf.Config["object_id"]
+	if ok {
+		a.objectID, ok = objectIDRaw.(string)
+		if !ok {
+			return nil, errors.New("could not convert 'object_id' config value to string")
+		}
+	}
+
+	clientIDRaw, ok := conf.Config["client_id"]
+	if ok {
+		a.clientID, ok = clientIDRaw.(string)
+		if !ok {
+			return nil, errors.New("could not convert 'client_id' config value to string")
+		}
+	}
```

Comment:
Should we also allow `mi_res_id` to be provided as well since we're making these additions anyways?

Reply:
I think I'll hold on adding `mi_res_id` for now. It's the least discoverable attribute from looking in the Azure portal. I also wonder if having 3 options to achieve effectively the same outcome might be more confusing than helpful. We can always add it if someone requests the option.